pkg/utils/namespace.go

```diff
@@ -0,0 +1,17 @@
+package utils
+
+import (
+	"io/ioutil"
+	"strings"
+)
+
+// Get the namespace that the Pod is running in.
+func PodNamespace() string {
+	if data, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace"); err == nil {
```

Comment:
Normally the pod namespace can be passed in use POD_NAMESPACE environment variable. See https://github.com/longhorn/longhorn/blob/47d38e621e0c52ae9c2a316baf0138a9b09b2fc6/deploy/longhorn.yaml#L739

This file seems to reflect `the default namespace to be used for namespaced API operations`. I am a bit unsure if it's the pod's running namespace. https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/#accessing-the-api-from-a-pod

Reply:
I updated it use the POD_NAMESPACE environment variable first and fall back to the file if POD_NAMESPACE isn't set.